cmd/api/src/api/v2/agi.go

```diff
@@ -430,6 +430,49 @@ func (s Resources) ListAssetGroupMembers(response http.ResponseWriter, request *
 	}
 }
 
+func (s Resources) ListAssetGroupMemberCountsByKind(response http.ResponseWriter, request *http.Request) {
+	var agMembers api.AssetGroupMembers
+
+	queryParameterFilterParser := model.NewQueryParameterFilterParser()
+	if queryFilters, err := queryParameterFilterParser.ParseQueryParameterFilters(request); err != nil {
+		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, api.ErrorResponseDetailsBadQueryParameterFilters, request), response)
+		return
+	} else {
+		for name, filters := range queryFilters {
+			if validPredicates, err := agMembers.GetValidFilterPredicatesAsStrings(name); err != nil || name == "primary_kind" {
```

Comment:
I wanted to point out the hardcode primary_kind, wasn't sure if there was a better way to prevent just this filter or not.

Reply:
We discussed in slack but the object being counted is the `AssetGroupMembers` so we can't make a new model as there's nothing to filter / count there and we didn't want to create a hidden dependency between the two models to keep in sync. Instead we opted to abstract the two to ensure these endpoints stay in-sync